### ISSUE TITLE
Add ability to easily retrieve the access_token

### ISSUE THREAD

**Issue by HarHarLinks** [KrazyKirby99999/simple-matrix-bot-lib#73]
**Is your feature request related to a problem? Please describe.**
I want to use access_token to log in, but obtain it with the bot itself.

**Describe the solution you'd like**
I would like to log in once by providing username + password on cli, retrieve a token, store it alongside the session, and reuse it next time instead my password.

**Describe alternatives you've considered**
Perhaps the token should be part of the session.txt anyway? https://spec.matrix.org/unstable/client-server-api/#relationship-between-access-tokens-and-devices

**Comment by HarHarLinks** [comment]
Hold on, it is in there

https://github.com/KrazyKirby99999/simple-matrix-bot-lib/blob/96b120e9ea7425a117fb08918a2993ec6fb86e41/simplematrixbotlib/auth.py#L93

So why `ValueError: password or login_token or access_token is required`? Because
https://github.com/KrazyKirby99999/simple-matrix-bot-lib/blob/96b120e9ea7425a117fb08918a2993ec6fb86e41/simplematrixbotlib/auth.py#L63-L70

Seems like an oversight, I think it should just `session_read_file()` in `__init__()` if neither of the above 3 arguments are set try to load from file and only fail if the file contains no access_token

**Comment by KrazyKirby99999** [comment]
Currently the access_token is encrypted before being stored inside of session.txt for security reasons. If you want to replace the username/password with the access_token after the first login, perhaps you could subclass the Creds class like the following:
```python
class TokenCreds(Creds):
     def __init__(self,
                 homeserver,
                 username=None,
                 password=None,
                 login_token=None,
                 access_token=None,
                 session_stored_file='session.txt'):
          
          if os.path.isfile("token.txt"):
               with open("token.txt", 'r') as f:
                    access_token = f.read()
          
          super().__init__(self, homeserver, username, password, login_token, access_token, session_stored_file)
     
     def session_write_file(self):
          super().session_write_file()
          with open("token.txt", 'w') as f:
               f.write(self.access_token)
```

Keep in mind that this is a quick idea off the top of my head that may not work. I am about to go to bed, so I cannot do much the moment, but in the next several days I will look at making your request possible.

**Comment by HarHarLinks** [comment]
Thanks for the hint, I have managed to extract the token by doing something like:

```python
class TokenCreds(Creds):
     def __init__(self, homeserver, username=None, password=None, login_token=None, access_token=None, session_stored_file='session.txt', device_name=f'matrix-jellyfin-bot {__version__}'):
         super().__init__(homeserver, username, password, login_token, access_token, session_stored_file)
         self.device_name = device_name

     def session_write_file(self) -> str:
         self._key = fw.key_from_pass(self.access_token)
         super().session_write_file()
         return self.access_token

async def initialSync(bot) -> str:
    from nio import SyncResponse
    await bot.api.login()
    resp = await bot.api.async_client.sync(timeout=65536, full_state=False)
    if isinstance(resp, SyncResponse):
        print("login and initial sync successful")
        bot.creds.access_token = bot.api.async_client.access_token
        return bot.creds.session_write_file()

creds = TokenCreds(homeserver=homeserver,
                   username=matrix_username,
                   password=matrix_password,
                   session_stored_file='session.txt')
bot = botlib.Bot(creds)
matrix_access_token = asyncio.get_event_loop().run_until_complete(initialSync(bot))
```

**Comment by KrazyKirby99999** [comment]
Glad that it works. I will work on a more intuitive method soon.

**Comment by HarHarLinks** [comment]
I don't think encrypting the device ID is necessary. Encrypting the token is nice, but the only case in which it is useful is if the user enters their password on app run so it is never stored on disk. I would argue that `session_stored_file` should be merged with `class Config`. It is probably nicer if the bot uses as few files as possible, which are currently session.txt, config.toml, a user config.file as long as `class Config` isn't extensible, and with e2e also an e2e sqlite3 DB.

If the user still wants to encrypt the access_token, there could be a flag for the Config, which encrypts the access_token but still saves it to the config.toml. Then on load the user would need to supply a password to load the session.

**Comment by KrazyKirby99999** [comment]
I agree with moving in that direction. My only concern with this is breaking compatibility with session.txt.

**Comment by KrazyKirby99999** [comment]
#113
